src/cursor/cur_std.c

```diff
@@ -1052,6 +1053,21 @@ __wt_cursor_reconfigure(WT_CURSOR *cursor, const char *config)
     } else
         WT_ERR_NOTFOUND_OK(ret, false);
 
+    /* Set the prefix search near flag. */
+    if ((ret = __wt_config_getones(session, config, "prefix_key", &cval)) == 0) {
+        /* Prefix search near configuration can only be used for row-store. */
```

Comment:
Something looks odd here - you do the checking regardless of whether the code goes on to F_CLR or F_SET the configuration option. Shouldn't the checking only happen when setting the option?